File: Source/root/src/main/java/root/communication/messages/Message.java
package root.communication.messages;

import java.util.Date;

public class Message {

	public MessageType type;

	public String username;
	public String roomName;

	public Date timestamp;

	public Message(MessageType type, Date timestamp) {
		this.type = type;
		this.timestamp = timestamp;
	}

	public void setOrigin(String username, String roomName) {
		this.username = username;
		this.roomName = roomName;
	}

}

File: Source/server/rabbit-game/src/SharedModel/Messages/Message.cs

namespace RabbitGameServer.SharedModel.Messages
{
	public class Message
	{
		public MessageType type { get; set; }

		public string username { get; set; }
		public string roomName { get; set; }

		// public DateTime timestamp {get; set;}

		public Message(MessageType type, string username, string roomName)
		{
			this.type = type;
			// this.timestamp = timestamp;

			this.username = username;
			this.roomName = roomName;
		}

	}
}